source/ranges.tex

```diff
@@ -3802,1398 +3855,2470 @@
-\tcode{pattern_} with \tcode{std::move(pattern)}.
+\effects Equivalent to: \tcode{return x.i_.current_ == y.i_.current_;}
 \end{itemdescr}
 
-\indexlibrary{\idxcode{split_view}!\idxcode{split_view}}%
+\indexlibrary{\idxcode{operator"!=}!\idxcode{split_view::inner_iterator}}%
 \begin{itemdecl}
-template<InputRange R, ForwardRange P>
-  requires Constructible<V, all_view<R>> &&
-           Constructible<Pattern, all_view<P>>
-constexpr split_view(R&& r, P&& p);
+friend constexpr bool operator!=(const inner_iterator& x, const inner_iterator& y)
+  requires ForwardRange<Base>;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\effects
-Initializes \tcode{base_} with \tcode{view::all(std::forward<R>(r))} and
-\tcode{pattern_} with \tcode{view::all(std\brk{}::forward<P>(p))}.
+\effects Equivalent to: \tcode{return !(x == y);}
 \end{itemdescr}
 
-\indexlibrary{\idxcode{split_view}!\idxcode{split_view}}%
+\indexlibrary{\idxcode{operator==}!\idxcode{split_view::inner_iterator}}%
 \begin{itemdecl}
-template<InputRange R>
-  requires Constructible<V, all_view<R>> &&
-           Constructible<Pattern, single_view<iter_value_t<iterator_t<R>>>>
-constexpr split_view(R&& r, iter_value_t<iterator_t<R>> e);
+friend constexpr bool operator==(const inner_iterator& x, default_sentinel_t);
+friend constexpr bool operator==(default_sentinel_t, const inner_iterator& x);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\effects
-Initializes \tcode{base_} with \tcode{view::all(std::forward<R>(r))} and
-\tcode{pattern_} with \tcode{single_view\{\brk{}std::move(e)\}}.
+\effects Equivalent to:
+\begin{codeblock}
+auto cur = x.i_.@\placeholder{current}@;
+auto end = ranges::end(x.i_.parent_->base_);
+if (cur == end) return true;
+auto [pcur, pend] = subrange{x.i_.parent_->pattern_};
+if (pcur == pend) return x.incremented_;
+do {
+  if (*cur != *pcur) return false;
+  if (++pcur == pend) return true;
+} while (++cur != end);
+return false;
+\end{codeblock}
 \end{itemdescr}
 
-\rSec3[range.split.outer]{Class template \tcode{split_view::outer_iterator}}
+\indexlibrary{\idxcode{operator"!=}!\idxcode{split_view::inner_iterator}}%
+\begin{itemdecl}
+friend constexpr bool operator!=(const inner_iterator& x, default_sentinel_t y);
+friend constexpr bool operator!=(default_sentinel_t y, const inner_iterator& x);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to: \tcode{return !(x == y);}
+\end{itemdescr}
+
+\indexlibrary{\idxcode{iter_swap}!\idxcode{split_view::inner_iterator}}%
+\begin{itemdecl}
+friend constexpr void iter_swap(const inner_iterator& x, const inner_iterator& y)
+  noexcept(noexcept(ranges::iter_swap(x.i_.@\placeholdernc{current}, y.i_.\placeholdernc{current}@)))
+  requires IndirectlySwappable<iterator_t<Base>>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to
+\tcode{ranges::iter_swap(x.i_.\placeholdernc{current}, y.i_.\placeholdernc{current})}.
+\end{itemdescr}
+
+\rSec3[range.split.adaptor]{\tcode{view::split}}
+
+\pnum
+The name \tc
```

Comment:
I'm not sure about normative wording in a under an overview heading, but locally this would be under the `view::elements [range.elements.adaptor]` heading. I don't know if it looks like this in R7, but R6 got this right.

Reply:
R7 has it under "overview".

Reply:
The issue apparently is that "N" would not be defined if the text was moved.

Reply:
Moving normative wording to under a descriptive heading for that reason seems weird. IIRC, elsewhere, the relation, if any, between _x_ and `x` is called out explicitly and not assumed. Here it doesn't happen. Wouldn't it be better to, in the description of `elements_view` above, to say "a `size_t N`" (like elsewhere it says "the type `T` of _E_") and replace _N<sub>th</sub>)_ with `N`? Apparently, normative wording describing `view::keys` and `view::values` was also lost in the process.

Reply:
"Overview" sections are, in fact, normative, as is the rest of the main body of the standard text (except for notes and footnotes).
Since this expressly changed between R6 and R7, I'm uneasy messing around here editorially. @CaseyCarter, can you comment? (Use the "Files changed" view to see the right context for these comments.)

Reply:
Aside: I've been trying to eradicate the archaic "introductions are semi-normative" belief when it pops up in LWG.

This text changed between R6 and R7 because (1) LWG was unhappy that the example uses `view::elements` so far from any description thereof, and (2) LWG dislikes the single-paragraph subclauses used to specify the range adaptor objects in general. We decided to merge the range adaptor object spec into this overview paragraph, and there was encouragement to editorially change the other range adaptor / range adaptor object specifications similarly.

Reply:
> Aside: I've been trying to eradicate the archaic "introductions are semi-normative" belief when it pops up in LWG.

For reference, that's in favour of?

Reply:
In favor of
> "Overview" sections are, in fact, normative, as is the rest of the main body of the standard text (except for notes and footnotes).

Reply:
@CaseyCarter: "there was encouragement to editorially change the other range adaptor / range adaptor object specifications similarly."

Please create an editorial issue for that, otherwise we'll forget.

Reply:
opened #3134